// vite-plugin-vue-devtools/dist/client/assets/DrawerRight.vue_vue_type_script_setup_true_lang-d7a700af.js
import { _ as _sfc_main$1 } from './IconButton.vue_vue_type_script_setup_true_lang-68ea38ae.js';
import { v as defineComponent, a$ as ref, cP as useElementSize, cf as onClickOutside, o as openBlock, c as createBlock, h as withCtx, e as createElementBlock, u as createVNode, x as createBaseVNode, f as renderSlot, k as mergeProps, z as unref, g as createCommentVNode, P as Transition } from './index-36a8f2c8.js';

const _hoisted_1 = {
  relative: "",
  "h-full": "",
  "w-full": "",
  "of-auto": ""
};
const __default__ = {
  inheritAttrs: false
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...__default__,
  __name: "DrawerRight",
  props: {
    modelValue: { type: Boolean },
    navbar: {},
    autoClose: { type: Boolean }
  },
  emits: ["close"],
  setup(__props, { emit }) {
    const props = __props;
    const el = ref();
    const { height: top } = useElementSize(() => props.navbar, void 0, { box: "border-box" });
    onClickOutside(el, () => {
      if (props.modelValue && props.autoClose)
        emit("close");
    }, {
      ignore: ["a", "button", "summary", '[role="dialog"]']
    });
    return (_ctx, _cache) => {
      const _component_VDIconButton = _sfc_main$1;
      return openBlock(), createBlock(Transition, {
        "enter-active-class": "duration-200 ease-in",
        "enter-from-class": "transform translate-x-1/1",
        "enter-to-class": "opacity-100",
        "leave-active-class": "duration-200 ease-out",
        "leave-from-class": "opacity-100",
        "leave-to-class": "transform translate-x-1/1"
      }, {
        default: withCtx(() => [
          _ctx.modelValue ? (openBlock(), createElementBlock("div", mergeProps({
            key: 0,
            ref_key: "el",
            ref: el,
            border: "l base",
            flex: "~ col gap-1",
            absolute: "",
            "bottom-0": "",
            "right-0": "",
            "z-10": "",
            "z-20": "",
            "of-auto": "",
            "text-sm": "",
            "glass-effect": "",
            style: { top: `${unref(top)}px` }
          }, _ctx.$attrs), [
            createVNode(_component_VDIconButton, {
              absolute: "",
              "right-2": "",
              "top-2": "",
              "z-20": "",
              "text-xl": "",
              icon: "carbon-close",
              onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("close"))
            }),
            createBaseVNode("div", _hoisted_1, [
              renderSlot(_ctx.$slots, "default")
            ])
          ], 16)) : createCommentVNode("", true)
        ]),
        _: 3
      });
    };
  }
});

/* Injected with object hook! */

export { _sfc_main as _ };
